xml/telnet_banners.xml

```diff
@@ -816,4 +816,680 @@
+  </fingerprint>
+  <fingerprint pattern="System administrator is connecting from">
+    <description>a DrayTek Vigor SOHO Router</description>
+    <!-- System administrator is connecting from 54.39.173.86\n\nReject the connection request !!! -->
+    <example _encoding="base64">
+      U3lzdGVtIGFkbWluaXN0cmF0b3IgaXMgY29ubmVjdGluZyBmcm9tIDU0LjM5LjE3My44NgoKUmVqZWN0IH
+      RoZSBjb25uZWN0aW9uIHJlcXVlc3QgISEh
+    </example>
+    <param pos="0" name="hw.vendor" value="Draytek"/>
+    <param pos="0" name="hw.device" value="Broadband router"/>
+    <param pos="0" name="hw.product" value="Vigor"/>
+  </fingerprint>
+  <fingerprint pattern=".*Version\s(\d*.\d*)\/OpenBSD.*">
+    <description>OpenBSD</description>
+    <!-- 220 killer09 FTP server (Version 6.4/OpenBSD/Linux-ftpd-0.17) ready. -->
+    <example _encoding="base64" os.version="6.4">
+      MjIwIGtpbGxlcjA5IEZUUCBzZXJ2ZXIgKFZlcnNpb24gNi40L09wZW5CU0QvTGludXgtZnRwZC0wLjE3KSByZWFkeS4K
+    </example>
+    <param pos="0" name="hw.vendor" value="OpenBSD"/>
+    <param pos="0" name="hw.family" value="OpenBSD"/>
+    <param pos="0" name="hw.product" value="OpenBSD"/>
+    <param pos="1" name="os.version"/>
+  </fingerprint>
+  <fingerprint pattern="FreeBSD\/([^\\s]+)\s+\(([^\s]+)\)">
+    <description>a FreeBSD</description>
+    <!-- FreeBSD/amd64 (ms.gymspgs.cz) (pts/0)\n\n\n\nlogin: -->
+    <example _encoding="base64" os.arch="amd64" host.name="ms.gymspgs.cz">
+      RnJlZUJTRC9hbWQ2NCAobXMuZ3ltc3Bncy5jeikgKHB0cy8wKQoKCgpsb2dpbjo=
+    </example>
+    <param pos="0" name="hw.vendor" value="FreeBSD"/>
+    <param pos="0" name="hw.family" value="FreeBSD"/>
+    <param pos="0" name="hw.product" value="FreeBSD"/>
+    <param pos="1" name="os.arch"/>
+    <param pos="2" name="host.name"/>
+  </fingerprint>
+  <fingerprint pattern="NetBSD">
+    <description>NetBSD</description>
+    <!-- NetBSD/evbsh3 (Fukuyama.Host_AKS_0555_WL-v2.60d) (ttyp1)  -->
+    <example _encoding="base64">
+      TmV0QlNEL21lc3NpbWlwcyAoKSAodHR5cDMpCgpsb2dpbjog
+    </example>
+    <param pos="0" name="hw.vendor" value="NetBSD"/>
+    <param pos="0" name="hw.family" value="NetBSD"/>
+    <param pos="0" name="hw.product" value="NetBSD"/>
+  </fingerprint>
+  <fingerprint pattern="IRIX\W\((.*)\)">
+    <description>SGI IRIX</description>
+    <!-- IRIX (artemis.biol.uoa.gr)\n\n\n\nlogin: -->
+    <example _encoding="base64" host.name="artemis.biol.uoa.gr">
+      SVJJWCAoYXJ0ZW1pcy5iaW9sLnVvYS5ncikKCgoKbG9naW46IA==
+    </example>
+    <param pos="0" name="hw.vendor" value="SGI"/>
+    <param pos="0" name="hw.family" value="IRIX"/>
+    <param pos="0" name="hw.product" value="IRIX"/>
+    <param pos="1" name="host.name"/>
+  </fingerprint>
+  <fingerprint pattern="^(?m)(RS [^\s]+) System Software, Version ([^\s]+).*">
+    <description>RiverStone router</description>
+    <!-- ++++++++++++++++++++++++++++++++++++++++\nRS 8000 System Software, Version 9.3.0.5\n
```

Comment:
If there really are `+` at the beginning of the banners, your regex will not match.

Reply:
Ah, the reason this works is because of the anchor in combination with multi-line:

```
  ^      assert start of string (or line, in multiline mode)
```